MapboxCoreNavigation/RouteController.swift

```diff
@@ -598,6 +575,32 @@ extension RouteController: CLLocationManagerDelegate {
         checkForFasterRoute(from: location)
     }
     
+    func monitorLegProgress(_ location: CLLocation) {
```

Comment:
We need to remove the word “monitor” from these methods. It means nothing and leads to the methods becoming kitchen sinks, because everything is a result of “monitoring”.

For now, `monitorStepProgress(_:)` should become `updateRouteStepProgress(for:)`, and this method should become `updateRouteLegProgress(for:)`. At some point, we should try to break up each method into more granular methods that can be unit tested more easily.

Reply:
Changed in https://github.com/mapbox/mapbox-navigation-ios/pull/1013/commits/24a2f6f6ce86f14877c164f296e35286cc0f713b